h2o-py/h2o/frame.py

```diff
@@ -126,7 +126,7 @@ def from_python(python_obj, destination_frame=None, header=0, separator=",", col
         assert_is_type(separator, I(str, lambda s: len(s) == 1))
         assert_is_type(column_names, None, [str])
         assert_is_type(column_types, None, [coltype])
-        assert_is_type(na_strings, None, [str])
+        assert_is_type(na_strings, None, [str], [[str]], { str: [str, None] })
```

Comment:
Sorry, not sure what I was thinking here. I'll have a look tomorrow.

The issue that brought me to this patch was that I was having values inferred (surprise) as NA via `water.parser.PreviewParseWriter.guessNAStrings`. I think I was using something like `{x: [None] for x in col_names}` to force this behavior to not happen. I'll try to confirm that just using the empty list in place of `[None]` is sufficient to suppress the guessing..